cmd/controller/main.go

```diff
@@ -89,14 +86,77 @@ func main() {
+
+	const numControllers = 1
+	cfg.QPS = numControllers * rest.DefaultQPS
+	cfg.Burst = numControllers * rest.DefaultBurst
+	opt := reconciler.NewOptionsOrDie(cfg, logger, stopCh)
+
+	kubeInformerFactory := kubeinformers.NewSharedInformerFactory(opt.KubeClientSet, opt.ResyncPeriod)
+	eventingInformerFactory := informers.NewSharedInformerFactory(opt.EventingClientSet, opt.ResyncPeriod)
+
+	subscriptionInformer := eventingInformerFactory.Eventing().V1alpha1().Subscriptions()
+	// TODO: remove unused after done integrating all controllers.
+	//deploymentInformer := kubeInformerFactory.Apps().V1().Deployments()
+	//coreServiceInformer := kubeInformerFactory.Core().V1().Services()
+	configMapInformer := kubeInformerFactory.Core().V1().ConfigMaps()
+
+	// Build all of our controllers, with the clients constructed above.
+	// Add new controllers to this array.
+	controllers := []*kncontroller.Impl{
+		subscription.NewController(
+			opt,
+			subscriptionInformer,
+		),
+	}
+	if len(controllers) != numControllers {
+		logger.Fatalf("Number of controllers and QPS settings mismatch: %d != %d", len(controllers), numControllers)
+	}
+
+	// Watch the logging config map and dynamically update logging levels.
+	opt.ConfigMapWatcher.Watch(logging.ConfigMapName(), logging.UpdateLevelFromConfigMap(logger, atomicLevel, logconfig.Controller))
+	// TODO: Watch the observability config map and dynamically update metrics exporter.
+	//opt.ConfigMapWatcher.Watch(metrics.ObservabilityConfigName, metrics.UpdateExporterFromConfigMap(component, logger))
+	if err := opt.ConfigMapWatcher.Start(stopCh); err != nil {
+		logger.Fatalw("failed to start configuration manager", zap.Error(err))
+	}
+
+	// Start all of the informers and wait for them to sync.
+	logger.Info("Starting informers.")
+	if err := kncontroller.StartInformers(
+		stopCh,
+		subscriptionInformer.Informer(),
+		configMapInformer.Informer(),
+	); err != nil {
+		logger.Fatalf("Failed to start informers: %v", err)
+	}
+
+	// Start all of the controllers.
+	logger.Info("Starting controllers.")
+	go kncontroller.StartAll(stopCh, controllers...)
+}
+
+// TODO: remove after done integrating all controllers.
+func startControllerRuntime(stopCh <-chan struct{}, cfg *rest.Config, logger *zap.SugaredLogger, atomicLevel zap.AtomicLevel) {
+	logger = logger.With(zap.String("controller/impl", "cr"))
+	logger.Info("Starting the controller")
+
 	kubeClient, err := kubernetes.NewForConfig(cfg)
 	if err != nil {
 		logger.Fatalf("Error building kubernetes clientset: %v", err)
 	}
 
 	// Watch the logging config map and dynamically update logging levels.
 	configMapWatcher := configmap.NewInformedWatcher(kubeClient, system.Namespace())
-	configMapWatcher.Watch(logconfig.ConfigMapName(), logging.UpdateLevelFromConfigMap(logger, atomicLevel, logconfig.Controller, logconfig.Controller))
+	configMapWatcher.Watch(logconfig.ConfigMapName(), logging.UpdateLevelFromConfigMap(logger, atomicLevel, logconfig.Controller))
```

Comment:
did I miss the resolution to why this is different than logging.ConfigMapName above on L125?

Reply:
this will be fixed as the work continues.

Reply:
fixed.